bm-persona/Libraries/LibraryDetailViewController.swift

```diff
@@ -0,0 +1,362 @@
+        button.topAnchor.constraint(equalTo: contentHelper.layoutMarginsGuide.topAnchor, constant: 400).isActive = true
+        //change the top anchor constant to 610 when including the card with usage information
+        button.leftAnchor.constraint(equalTo: scrollView.layoutMarginsGuide.leftAnchor).isActive = true
+        button.rightAnchor.constraint(equalTo: scrollView.layoutMarginsGuide.rightAnchor).isActive = true
+        button.heightAnchor.constraint(equalToConstant: 50).isActive = true
+        button.setTitle("Book a Study Room >", for: .normal)
+        button.titleLabel!.font = Font.semibold(14)
+        button.titleLabel!.textColor = .white
+        
+    }
+    
+    func setUpScrollView() {
+        scrollView = UIScrollView(frame: .zero)
+        scrollView.layoutMargins = UIEdgeInsets(top: 10, left: 10, bottom: 0, right: 10)
+        scrollView.alwaysBounceVertical = true
+        view.addSubview(scrollView)
+        scrollView.translatesAutoresizingMaskIntoConstraints = false
+        scrollView.topAnchor.constraint(equalTo: view.layoutMarginsGuide.topAnchor).isActive = true
+        scrollView.leftAnchor.constraint(equalTo: view.layoutMarginsGuide.leftAnchor).isActive = true
+        scrollView.rightAnchor.constraint(equalTo: view.layoutMarginsGuide.rightAnchor).isActive = true
+        scrollView.bottomAnchor.constraint(equalTo: view.layoutMarginsGuide.bottomAnchor).isActive = true
+        
+        contentHelper = UIView()
+        scrollView.addSubview(contentHelper)
+    }
+}
+
+extension LibraryDetailViewController {
+    
+    @objc func toggleFave(sender: UIButton) {
+               if library!.isFavorited {
+                sender.setImage(#imageLiteral(resourceName: "not-favorited-icon"), for: .normal)
+                library!.isFavorited = false
+               } else {
+                sender.setImage(#imageLiteral(resourceName: "favorited-icon"), for: .normal)
+                library!.isFavorited = true
+        }
+    }
+    
+}
+    
+    extension  LibraryDetailViewController : CLLocationManagerDelegate {
+        
+        func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+            let userLocation : CLLocation = locations[0] as CLLocation
+            userCoords.latitude = userLocation.coordinate.latitude
+            userCoords.longitude = userLocation.coordinate.longitude
+            DispatchQueue.main.async {
+                //reloads card with distance marker
+                if self.library != nil, let libLat = self.library!.latitude, let libLong = self.library!.longitude {
+                    let dist = self.calcDistance(lib_Lat: libLat, lib_Long: libLong)
+                    self.distLabel.text = String(format: "%.1f", dist) + " miles away"
+                    
+                } else {
+                    self.distLabel.text = "distance not availible"
+                }
+            }
```

Comment:
same here, want to use HasLocation protocol's getDistanceToUser function instead